src/amount.rs

```diff
@@ -87,63 +88,105 @@ pub type AmountCounter = u32;
 ///
 #[derive(Clone, Debug, Copy, Default, Serialize, Deserialize)]
 pub struct Amount {
-    pub count: AmountCounter,
-    pub unit: PowerOfTen,
+    count: AmountCounter,
+    unit: PowerOfTen,
 }
 
 /// A NormalizedAmount is just like an Amount except that count is a Big Integer.
 /// So sum or difference of any two Amounts sharing the same unit can be represented
 /// with a NormalizedAmount.
 ///
 /// For now, have this only for internal use/ops.
-#[derive(Debug)]
+#[derive(Debug, Default)]
 struct NormalizedAmount {
     count: Integer,
     unit: PowerOfTen,
 }
 
 impl Amount {
-    pub fn new(count: AmountCounter, unit: PowerOfTen) -> Self {
-        // We constrain count to ::counter_max().  If you want to use a bigger value,
-        // you must change the unit.
-        debug_assert!(count <= Self::counter_max());
+    /// maximum value for Self::count. MUST equal 10^COUNT_MAX_TEN_POW
+    //  note: we define these separately to avoid unnecessary runtime 10.pow() call.
```

Comment:
[`pow`](https://doc.rust-lang.org/nightly/std/primitive.i64.html#method.pow) is a const function so 10.pow() willl be evaluated at compile time, it shouldn't impact runtime

Reply:
doing that now.